sdk/core/azure-core/src/main/java/com/azure/core/implementation/ByteBufferCollector.java

```diff
@@ -60,7 +60,7 @@ public ByteBufferCollector(int initialSize) {
      * @throws IllegalStateException If the size of the backing array would be larger than {@link Integer#MAX_VALUE}
      * when the passed buffer is written.
      */
-    public synchronized void write(ByteBuffer byteBuffer) {
+    public void write(ByteBuffer byteBuffer) {
```

Comment:
Why isn't this `synchronized` now? Don't we need to update the `position` in a thread-safe way?